app/src/pages/inside/common/expandedOptions/expandedOptions.tsx

```diff
@@ -61,6 +76,45 @@ export const ExpandedOptions = ({
 }: ExpandedOptionsProps) => {
   const { formatMessage } = useIntl();
 
+  const storageKey = instanceKey ? `${BASE_STORAGE_KEY}_${instanceKey}` : BASE_STORAGE_KEY;
+
+  const [expandedIds, setExpandedIds] = useState<number[]>(() => {
+    try {
+      const parsed = getStorageItem(storageKey) as number[] | null;
+
+      if (!parsed) return [];
+
+      if (Array.isArray(parsed) && parsed.every((id) => typeof id === 'number')) {
+        return parsed;
+      }
+
+      return [];
+    } catch {
+      return [];
+    }
+  });
+
+  const onToggleFolder = useCallback(
+    (folder: TransformedFolder) => {
+      setExpandedIds((prevIds) => {
+        const isExpanded = prevIds.includes(folder.id);
+        let newIds: number[];
+
+        if (isExpanded) {
+          const idsToRemove = getFolderAndDescendantIds(folder);
+          newIds = prevIds.filter((id) => !idsToRemove.includes(id));
```

Comment:
Please don't forget to add empty line after variables block
```suggestion
          const idsToRemove = getFolderAndDescendantIds(folder);

          newIds = prevIds.filter((id) => !idsToRemove.includes(id));
```